static/js/apps/base/main.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react";
+import ReactDOM from "react-dom";
+
+import { loadLocaleData } from "../../i18n/i18n";
+import { FooterMenu, HeaderMenu } from "../../shared/types/base";
+import { FooterApp } from "./footerApp";
+import { HeaderApp } from "./headerApp";
+import { getLabels, getRoutes } from "./utilities/utilities";
+
+window.addEventListener("load", (): void => {
+  loadLocaleData("en", [import("../../i18n/compiled-lang/en/units.json")]).then(
+    () => {
+      renderPage();
+    }
+  );
+});
+
+function renderPage(): void {
+  const headerMenu = JSON.parse(
+    document.getElementById("metadata-base").dataset.header
+  ) as HeaderMenu[];
+  const footerMenu = JSON.parse(
+    document.getElementById("metadata-base").dataset.footer
+  ) as FooterMenu[];
+
+  const name = document.getElementById("metadata-base").dataset.name;
+  const logoPath = document.getElementById("metadata-base").dataset.logoPath;
+  const hideFullFooter =
+    document.getElementById("metadata-base").dataset.hideFullFooter === "true";
```

Comment:
This might need to get lowercased-- im seeing the full footer in other pages in the app: 

in http://localhost:8080/tools/statvar

```
<div id="metadata-base" class="d-none"
...
  data-hide-full-footer="True"
...
/>
```